### PR TITLE
Add Glonass support into signal.c/h

### PR DESCRIPTION
**Pull request by pmiettinen**
Depends on
https://github.com/swift-nav/libswiftnav/pull/325

### CODE REVIEW

File: src/signal.h [hunk 1]
```diff
@@ -20,29 +21,37 @@
...
+#define CODE_GLO_L1CA_SUPPORT     0
+#define CODE_GLO_L2CA_SUPPORT     0
 /* =================================== */
 
 /** \addtogroup signal
  * \{ */
 
 /* Number of signals on each code supported
  * on the current hardware platform. */
-#define PLATFORM_SIGNAL_COUNT_GPS_L1CA      (CODE_GPS_L1CA_SUPPORT ?          \
```
Comment (ljbade) [inline src/signal.h]: Is the change in spaces to the `\` intentional?

**Comment by ljbade** [comment]
Other than spacing changes (see comment), LGTM.

File: src/signal.h (hunk 1 above)
Comment (pmiettinen) [inline src/signal.h]: Yes, I fitted them inside 80 char rule. Was there some specific purpose them being that far?

Comment (ljbade) [inline src/signal.h]: Not sure, @jacobmcnamee might know

**Comment by pmiettinen** [comment]
@ljbade libswiftnav submodule needs to be updated to support the change, correct?

**Comment by ljbade** [comment]
@pmiettinen Yes

**Comment by ljbade** [comment]
Update it to libswiftnav master

**Comment by pmiettinen** [comment]
@ljbade Done and travis build is OK. Do we still wait for @jacobmcnamee 's comment?

**Comment by ljbade** [comment]
Yeah, I'd give him a chance to look. otherwise it appears it is ready to merge

File: src/signal.h [hunk 2]
```diff
@@ -20,29 +21,37 @@
...
+                                             NUM_SIGNALS_GLO_L2CA : 0)
 
 /* Number of signals on each constellation supported
  * on the current hardware platform. */
-#define PLATFORM_SIGNAL_COUNT_GPS     (PLATFORM_SIGNAL_COUNT_GPS_L1CA +       \
+#define PLATFORM_SIGNAL_COUNT_GPS     (PLATFORM_SIGNAL_COUNT_GPS_L1CA +\
                                        PLATFORM_SIGNAL_COUNT_GPS_L2CM)
 #define PLATFORM_SIGNAL_COUNT_SBAS    (PLATFORM_SIGNAL_COUNT_SBAS_L1CA)
+#define PLATFORM_SIGNAL_COUNT_GLO     (PLATFORM_SIGNAL_COUNT_GLO_L1CA +\
+                                       PLATFORM_SIGNAL_COUNT_GLO_L1CA)
```
Comment (jacobmcnamee) [inline src/signal.h]: PLATFORM_SIGNAL_COUNT_GLO_L2CA

File: src/signal.h (hunk 1 above)
Comment (jacobmcnamee) [inline src/signal.h]: They were right at 80 before. That avoids the inevitable problem of having to push them out later when a longer define is added.

File: src/signal.h [hunk 3]
```diff
@@ -20,29 +21,37 @@
...
 
 /* Total number of signal supported
  * on the current hardware platform. */
-#define PLATFORM_SIGNAL_COUNT         (PLATFORM_SIGNAL_COUNT_GPS +            \
+#define PLATFORM_SIGNAL_COUNT         (PLATFORM_SIGNAL_COUNT_GPS +     \
```
Comment (jacobmcnamee) [inline src/signal.h]: Add PLATFORM_SIGNAL_COUNT_GLO

**Comment by jacobmcnamee** [comment]
Added a couple inline comments.

Let's merge this first, but for reference I have a PR to move the `/* Platform-specific code support ==== */` stuff into an actual platform-specific header file.

File: src/signal.h (hunk 1 above)
Comment (pmiettinen) [inline src/signal.h]: Yep, looking closer I noticed that Geany editor has the default line ending mark at column 72 for some reason... Sorry.

**Comment by pmiettinen** [comment]
@ljbade @jacobmcnamee Good catches on those few oversights. Mistakes are now fixed and we should be good to merge.

**Comment by ljbade** [comment]
Looks good. Merging.

**Pull request merged**